drop-transfer/src/storage_dispatch.rs

```diff
@@ -20,14 +20,24 @@ impl<'a> StorageDispatch<'a> {
 
     pub async fn handle_event(&mut self, event: &crate::Event) {
         match event {
-            crate::Event::FileUploadStarted(transfer, file_id) => {
+            crate::Event::FileUploadStarted(transfer, file_id, bytes) => {
+                self.store_progres(transfer.id(), file_id, *bytes as _);
                 self.storage
-                    .insert_outgoing_path_started_state(transfer.id(), file_id.as_ref())
+                    .insert_outgoing_path_started_state(
+                        transfer.id(),
+                        file_id.as_ref(),
+                        *bytes as _,
```

Comment:
What does this do in Rust?

Reply:
Do you mean `as _` it's the same as `as i64` but the compiler can infer the type because of the argument type of the function

Reply:
oh okay, thanks